samples/snippets/csharp/programming-guide/classes-and-structs/local-functions-async1.cs

```diff
@@ -0,0 +1,35 @@
+using System;
+using System.Threading.Tasks;
+
+class Example
+{
+   static void Main()
+   {
+      int result = GetMultipleAsync(6).Result;
+      Console.WriteLine($"The returned value is {result:N0}");
+   }
+
+   static async Task<int> GetMultipleAsync(int secondsDelay)
+   {
+      Console.WriteLine("Executing GetMultipleAsync...");
+      if (secondsDelay < 0 || secondsDelay > 5)
+         throw new ArgumentOutOfRangeException("secondsDelay cannot exceed 5.");
+         
+      await Task.Delay(secondsDelay * 1000);
```

Comment:
I personally prefer to use the `TimeSpan` overload to `Delay()`, especially in example code, since I think it makes the code cleaner and clearer. I.e.: `await Task.Delay(TimeSpan.FromSeconds(secondsDelay));`. But it's just a personal preference.